pkg/statsd/statsd_test.go

```diff
@@ -82,7 +83,20 @@ func (cb *countingBackend) Name() string {
 }
 
 func (cb *countingBackend) SendMetricsAsync(ctx context.Context, m *gostatsd.MetricMap, callback gostatsd.SendCallback) {
-	atomic.AddUint64(&cb.metrics, uint64(m.NumStats))
+	count := 0
```

Comment:
This does not count the same value as the original.  This counts the number of tagsets, the original counted the number of datapoints.  As such, the TestStatsdThroughput summary looks worse.